cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/resource/AbstractOperations.java

```diff
@@ -165,4 +167,68 @@ public T get(String namespace, String name) {
         return operation().inNamespace(namespace).withLabels(labels).list().getItems();
     }
 
+    /**
+     * Waits until resource is in the Ready state
+     *
+     * @param namespace The namespace
+     * @param name The resource name
+     * @param timeout Timeout in {@code timeUnit}
+     * @param timeUnit Time unit
+     */
+    public Future<Void> waitUntilReady(String namespace, String name, long timeout, TimeUnit timeUnit) {
+        Future<Void> fut = Future.future();
+        log.info("Waiting for {} resource {} in namespace {} to get ready", resourceKind, name, namespace);
+        AtomicInteger calls = new AtomicInteger(0);
+        long timer = 1000L;
+        long timeoutInMs = timeUnit.toMillis(timeout);
+
+        vertx.setPeriodic(timer, timerId -> {
+            calls.addAndGet(1);
+
+            vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
+                    future -> {
+                        try {
+                            R resourceOp = operation().inNamespace(namespace).withName(name);
+                            T resource = resourceOp.get();
+                            if (resource != null)   {
+                                if (Readiness.isReadinessApplicable(resource)) {
+                                    Boolean ready = resourceOp.isReady();
+
+                                    if (ready == true) {
+                                        future.complete();
+                                    } else {
+                                        future.fail("Not ready yet");
+                                    }
+                                } else {
+                                    future.complete();
+                                }
+                            } else {
+                                log.warn("{} {} in namespace {} doesn't exist", resourceKind, name, namespace);
+                                future.fail("Resource doesn't exist");
+                            }
+                        }
+                        catch (Exception e) {
+                            log.warn("Caught exception while waiting for {} {} in namespace {} to get ready", resourceKind, name, namespace, e);
+                            future.fail(e);
+                        }
+                    },
+                    false,
+                    res -> {
+                        if (res.succeeded())    {
+                            vertx.cancelTimer(timerId);
+                            log.info("{} {} in namespace {} is ready", resourceKind, name, namespace);
+                            fut.complete();
+                        } else {
+                            if (calls.get() * timer > timeoutInMs)   {
```

Comment:
Why not `long t0 = System.currentTimeMillis();` outside the call to `setPeriodic()` and then timeout when `System.currentTimeMills() - t0 > timeoutInMs`? That avoids needing an atomic count of the number of calls. If you really want to do it using `calls.get() * timer` I think you could just declare `volatile int calls`, rather than using an `AtomicInteger`